### PR TITLE
Fix type checking issue with `model_fields` and `model_computed_fields`

### PR DESCRIPTION
**Pull request by sydney-runkle**
Fix https://github.com/pydantic/pydantic/issues/10907

### CODE REVIEW

**Comment by Viicos** [comment]
> * Thus, I think decorating with the `classonlyproperty` is a bit misleading.

The naming of the decorator can be improved, but with this approach, accessing the property _does_ work at runtime. However, it will emit a runtime deprecation warning and a type checker error/warning

**Comment by Viicos** [comment]
Sounds good, could you please add some tests in the `typechecking` folder? Probably in `base_model.py`.

**Comment by sydney-runkle** [comment]
Oh so odd, I think I have them locally but didn't push...

File: tests/typechecking/base_model.py [hunk 1]
```diff
@@ -28,3 +28,6 @@ class Knight(BaseModel):
 
 
 k = Knight()  # type: ignore[call-arg]  # pyright: ignore[reportCallIssue]
+
+MyModel.model_fields.keys()
+MyModel.model_computed_fields.keys()
```
Comment (Viicos) [inline tests/typechecking/base_model.py:33]:
```suggestion
assert_type(Knight.model_fields, dict[str, FieldInfo])
assert_type(Knight.model_computed_fields, dict[str, ComputedFieldInfo])
assert_type(k.model_fields, dict[str, FieldInfo])
assert_type(k.model_computed_fields, dict[str, ComputedFieldInfo])
```

importing `assert_type` from `typing_extensions`

**Pull request merged**